### PR TITLE
Add BanTimeSeconds to Network

### PR DESCRIPTION
**Pull request by fassadlr**
This is now a base configurable value. 

We also assert that the value is less than MaxReorg * TargetSpacing / 2.

### CODE REVIEW

**Comment by fassadlr** [comment]
https://dev.azure.com/Stratisplatformuk/StratisBitcoinFullNode/_workitems/edit/4023

File: src/Stratis.Bitcoin.Networks/StratisMain.cs [hunk 1]
```diff
@@ -52,6 +52,7 @@ public StratisMain()
             this.DefaultConfigFilename = StratisDefaultConfigFilename;
             this.MaxTimeOffsetSeconds = 25 * 60;
             this.CoinTicker = "STRAT";
+            this.BanTimeSeconds = 16000; // 500 (MaxReorg) * 64 (TargetSpacing) / 2
```
Comment (quantumagi) [inline src/Stratis.Bitcoin.Networks/StratisMain.cs]: `// 500 (MaxReorg) * 64 (TargetSpacing) / 2 = 4 hours, 26 minutes and 40 seconds`

File: src/NBitcoin/Network.cs [hunk 2]
```diff
@@ -64,6 +64,14 @@ public abstract class Network
     {
         protected Block Genesis;
 
+        /// <summary>
+        /// The default amount of seconds to keep misbehaving peers from reconnecting.
+        /// <para>
+        /// This value should be calculated as (TargetSpacingSeconds * maxReorgLength) / 2.
+        /// </para>
+        /// </summary>
+        public int BanTimeSeconds { get; protected set; }
```
Comment (quantumagi) [inline src/NBitcoin/Network.cs]: `DefaultBanTimeSeconds`

Comment (quantumagi) [reply src/NBitcoin/Network.cs]: Change the name to prevent misuse of the default value when it may  have been overridden from the command line.

Comment (quantumagi) [reply src/NBitcoin/Network.cs]: Done.

File: src/Stratis.Bitcoin.Networks/StratisMain.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.Networks/StratisMain.cs]: Done.

**Comment by fassadlr** [comment]
Thanks for taking this over @quantumagi :)

**Pull request merged**